integration/spark2/src/main/scala/org/apache/spark/sql/execution/CastExpressionOptimization.scala

```diff
@@ -66,6 +68,19 @@ object CastExpressionOptimization {
     }
   }
 
+  def typeCastStringToLongForDateType(v: Any): Any = {
+    try {
+      // spark also uses castToTimestamp only to convert  time to long.So to syn with spark ,
+      // Filter cast format should be same so used castToTimestamp method .
+      // Spark uses it in Cast.scala under ConstantFolding Rule before carbon optimizer)
+      val value = DateTimeUtils.stringToTimestamp(UTF8String.fromString(v.toString)).get
```

Comment:
When parse fails it gives none, none case behaviour needs to validate as per hive, whether null values are considered in output or not.